deepspeed/ops/transformer/inference/op_binding/mlp_gemm.py

```diff
@@ -12,18 +13,34 @@ class MLPGemmOp(BaseOp):
 
     def __init__(self, config: DeepSpeedInferenceConfig):
         super(MLPGemmOp, self).__init__(config)
-        if self.config.fp16:
-            self.mlp_gemm_func = self.inference_cuda_module.mlp_gemm_fp16
-        else:
-            self.mlp_gemm_func = self.inference_cuda_module.mlp_gemm_fp32
+        try:
+            if self.config.fp16:
+                self.mlp_gemm_func = self.inference_module.mlp_gemm_fp16
+            elif self.config.bf16:
+                self.mlp_gemm_func = self.inference_module.mlp_gemm_bf16
+            else:
+                self.mlp_gemm_func = self.inference_module.mlp_gemm_fp32
+        except AttributeError:
+            self.mlp_gemm_func = None
 
     def forward(self, input: torch.Tensor, residual: torch.Tensor, input_bias: torch.Tensor,
                 weight_interm: torch.Tensor, weight_out: torch.Tensor, bias: torch.Tensor, gamma: torch.Tensor,
                 beta: torch.Tensor):
-        output, residual_add = self.mlp_gemm_func(
-            input, residual, input_bias, weight_interm, weight_out, bias, gamma, beta, self.config.epsilon,
-            self.config.pre_layer_norm, self.config.mlp_after_attn,
-            weight_interm.scale if hasattr(weight_interm, 'scale') else torch.empty(1),
-            weight_out.scale if hasattr(weight_out, 'scale') else torch.empty(1), self.config.q_int8,
-            self.config.mlp_act_func_type, self.config.transposed_mode)
+        if self.mlp_gemm_func != None:
+            output, residual_add = self.mlp_gemm_func(
+                input, residual, input_bias, weight_interm, weight_out, bias, gamma, beta, self.config.epsilon,
+                self.config.pre_layer_norm, self.config.mlp_after_attn,
+                weight_interm.scale if hasattr(weight_interm, 'scale') else torch.empty(1),
+                weight_out.scale if hasattr(weight_out, 'scale') else torch.empty(1), self.config.q_int8,
+                self.config.mlp_act_func_type, self.config.transposed_mode)
+        else:
+            # fallback
```

Comment:
Need to have a mechanism to enforce going through fallback path to test fallback path on cuda device.  Consider use an environment variable 'DS_KI_ENFORCE_FALLBACK'

Reply:
Use an environment variable DS_FORCE_KI_FALLBACK better be in a seperate PR.